mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td

```diff
@@ -1883,7 +1883,7 @@ def Tosa_PadOp : Tosa_InferShapedTypeOp<"pad"> {
   let arguments = (ins
     Tosa_RankedTensor:$input1,
     Tosa_Shape:$padding,
-    Optional<Tosa_Rank0Tensor>:$pad_const,
+    Optional<Tosa_ScalarTensor>:$pad_const,
```

Comment:
Probably another patch is needed here as well to fix optionality?

Reply:
Yes. That's on the way.